app/grandchallenge/notifications/templates/notifications/notification_list.html

```diff
@@ -45,60 +46,65 @@ <h2>Notifications</h2>
-                                {% elif object.action.actor_content_type.model == 'algorithmpermissionrequest' or object.action.actor_content_type.model == 'readerstudypermissionrequest' or object.action.actor_content_type.model == 'archivepermissionrequest' %}
-                                    Your registration request for <a href="{{ object.action.actor.base_object.get_absolute_url }}">{{ object.action.actor.object_name }}</a>
-                                {% elif object.action.actor_content_type.model == 'registrationrequest' %}
-                                    Your participation request for <a href="{{ object.action.actor.challenge.get_absolute_url }}">{{ object.action.actor.challenge }}</a>
-                                {% elif object.action.actor_content_type.model == 'evaluation' and object.action.actor.submission.creator == request.user %}
-                                    {% if object.action.verb == "failed" %}
-                                        Your <a href="{{ object.action.actor.submission.get_absolute_url }}">submission</a> to <a href="{{ object.action.target.challenge.get_absolute_url }}">{{ object.action.target.challenge.short_name }}</a> failed. <span class="text-truncate font-italic text-muted align-middle mx-2 ">{{ object.action.actor.error_message }}</span>
-                                    {% else %}
-                                        Your <a href="{{ object.action.actor.submission.get_absolute_url }}">submission</a> to <a href="{{ object.action.target.challenge.get_absolute_url }}">{{ object.action.target.challenge.short_name }}</a> was successful.
-                                    {% endif %}
-                                {% elif object.action.actor_content_type.model == 'evaluation' and object.action.target.creator != request.user %}
-                                    {% if object.action.verb == "failed" %}
-                                       The <a href="{{ object.action.actor.submission.get_absolute_url }}">submission</a> from {{ object.action.actor.submission.creator|user_profile_link }} to <a href="{{ object.action.target.challenge.get_absolute_url }}">{{ object.action.target.challenge.short_name }}</a> failed. <span class="text-truncate font-italic text-muted align-middle mx-2 ">{{ object.action.actor.error_message }}</span>
-                                    {% else %}
-                                        There is a new <a href="{{ object.action.actor.submission.get_absolute_url }}">result</a> for <a href="{{ object.action.target.challenge.get_absolute_url }}">{{ object.action.target.challenge.short_name }}</a> from {{ object.action.actor.submission.creator|user_profile_link }}.
+                            {% if object.type != "GENERIC" %}
+                                {% print_notification notification=object user=request.user as current_notification %}
+                                {{ current_notification }}
+                            {% else %}
```

Comment:
Everything within the else statement here can go after we are done transferring the old notificiations to the new format. The template will be much cleaner!

Reply:
That's really great!